packages/postcss-discard-comments/src/index.js

```diff
@@ -41,7 +41,7 @@ function pluginCreator(opts = {}) {
 
     const result = space(parsed).join(' ');
 
-    replacerCache[key] = result;
+    replacerCache.set(key, result);
 
     return result;
   }
```

Comment:
Ideally we can use WeakMap here and store AST nodes, so it reduces memory usage,  but let's think about it late

Reply:
I don't think storing the AST nodes in a WeakMap would be equivalent, because right now it is returns a cached result for different nodes with the same structure, while if we use the node as a key it will only return a cached result if it visits the same node again. In fact since the plugin uses `OnceExit` it should not see the same node twice, so using the AST node as a cache key would never return a cache hit.